lib/routes/account.js

```diff
@@ -891,12 +893,37 @@ module.exports = function (
       handler: function (request, reply) {
         log.begin('Account.RecoveryEmailStatus', request)
         var sessionToken = request.auth.credentials
-        reply(
-          {
-            email: sessionToken.email,
-            verified: sessionToken.emailVerified
+
+        cleanUpIfAccountInvalid()
+          .then(createResponse)
+          .done(reply, reply)
+
+        function cleanUpIfAccountInvalid() {
+          // Some historical bugs mean we've allowed creation
+          // of accounts with invalid email addresses. These
+          // can never be verified, so the best we can do is
+          // to delete them so the browser will stop polling.
+          if (!sessionToken.emailVerified) {
+            if (!validators.isValidEmailAddress(sessionToken.email)) {
+              return db.deleteAccount(sessionToken)
+                .then(
+                  function () {
+                    // Act as though we deleted the account asynchronously
+                    // and caused the sessionToken to become invalid.
+                    throw error.invalidToken()
+                  }
```

Comment:
Related browser behaviour bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1226449

I tested in release and nightly by signing up with a new account, then deleting it before verifying.  Release just stayed in the "you need to verify your email" state, and I'm not sure whether it stopped polling.  Nightly discovered the deletion and reverted to "sign in the sync" state.  Both of these seem OK and are no worse than current behaviour.

I *believe* the behaviour should be the same as what you'd get if you entered an email that bounced, because you'd get the same effect - the account is deleted and your session token becomes invalid.